app/controllers/volunteers_controller.rb

```diff
@@ -187,24 +183,17 @@ def super_admin
 
   def region_admin
     @regions = Region.all
+    @admin_region_ids = current_volunteer.admin_region_ids
     if current_volunteer.super_admin?
       @my_admin_regions = @regions
       @my_admin_volunteers = Volunteer.all
     else
-      @my_admin_regions = current_volunteer.assignments.collect do |assignment|
-         assignment.admin ? assignment.region : nil
-      end.compact
-
-      admin_region_ids = @my_admin_regions.collect { |my_admin_region| my_admin_region.id }
-
       @my_admin_volunteers = Volunteer.all.collect do |volunteer|
         ((volunteer.regions.length == 0) ||
-        (admin_region_ids & volunteer.regions.collect { |region| region.id }).length > 0) ? volunteer : nil
+        (@admin_region_ids & volunteer.regions.pluck(:id)).length > 0) ? volunteer : nil
```

Comment:
Why wouldn't you use the the join here: https://github.com/boulder-food-rescue/food-rescue-robot/blob/master/app/controllers/volunteers_controller.rb#L62

Reply:
Great question!

Reply:
That was written way after I poked at this PR, I think. 

I'll have to redo now after we worked on / merged things today!